### PR TITLE
feat: add env package

### PR DESCRIPTION
**Pull request by segunadebayo**
## 📝 Description

This PR introduces a new package `@chakra-ui/env` that provides the correct `window` and `document` object globally to all components and hooks.

It leverages the ref callback in react to determine the owner window and document using a temporarily rendered `span` element.

Let's say we need to design a portal component, we can leverage the `EnvironmentProvider` and `useEnvironment`

```jsx
// 1. create the portal that appends to the env's `document.body`

const Portal = ({ children }) => {
  const env = useEnvironment()
  return createPortal(children, env.document.body)
}

// 2. Render the portal in an iframe to see that the portal automatically appends to the iframe's
// `document.body`, not the root window

 <IFrame style={{ background: "yellow" }}>
        <EnvironmentProvider>
          <Portal>Inside iframe</Portal>
        </EnvironmentProvider>
 </IFrame>
```

My goals:
- Add `EnvironmentProvider` within `ChakraProvider` so that users don't have to think about it.
- Within our hooks where have `useOutsideClick` or `useWindowSize`, we can leverage the `useEnvironment` to gain access to the correct `window` or `document` instead of using the root window or document.

### CODE REVIEW

**Review by with-heart — APPROVED** [review]
This functionality definitely makes perfect sense and is a solid improvement! 

One thing I'm wondering is if this belongs directly within `@chakra-ui/react` or if it should be named `@chakra-ui/react-env` or something. Assuming we'll eventually be merging the different frameworks under one roof, I feel like it's important that we try to slowly move in a direction where framework-specific things are named in a way that makes that clear.

File: packages/env/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# @chakra-ui/env
+
+A Quick description of the component
```
[the remark this answers is not in the document]
Comment (segunadebayo) [reply packages/env/README.md:3]: That makes sense! I'll rename the package to `react-env` and add some description in there.

File: packages/env/tests/env.test.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+describe("useEnv", () => {
+  test("it works", () => {
+    expect(true).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (segunadebayo) [reply packages/env/tests/env.test.tsx:3]: Hahaha. I'll add some tests shortly

File: packages/env/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# @chakra-ui/react-env
```
Comment (with-heart) [inline packages/env/README.md:1]: Perfect! Descriptive name and descriptive `README`. We're golden!

**Pull request merged**